ironfish/src/network/peerNetwork.ts

```diff
@@ -840,6 +854,18 @@ export class PeerNetwork {
       return false
     }
 
+    const hash = new Transaction(message.message.transaction).hash()
+
+    const peersToSendTo = this.getPeersToSendTo(message.peerIdentity, hash)
+
+    if (
```

Comment:
This is the important change here. It replaces the gossip nonce filter for transactions. The nonce filter guards against 3 actions:
1. We don't sync a transaction gossip message to accounts twice (this includes the expensive operation of verifying and validating the transaction)
2. We don't try to add the same gossip message to our mempool twice (this also includes verifying and validating the transaction)
3. We don't send the same gossip message to our peers twice

This check is basically guarding against all those things so that we can remove the checking of the gossip nonce for transactions

Reply:
What's the trade offs of doing this? We might starve the network of deposits because rebroadcasting doesn't work? You basically get 1 shot to broadcast your deposit and thats it right?

Reply:
The `await this.node.accounts.transactionIsSynced(hash)` is currently not implemented. That is one thing I did not know how to do. Worst case scenario we could keep a in memory filter in the `PeerNetwork` for transaction hashes that have already been synced

Reply:
Yea that's right. This was the proposed change because both ETH and BTC are implemented this way. We do have a couple guarantees. A peer is not getting the transaction a second time if
1. We already successfully sent them the transaction
2. They sent us the transaction

I think this is good reason not to send it to that peer again but if you think we should allow sending a transaction to those peers again then I'm open to talking about it

Reply:
Bitcoin and Ethereum both filter outgoing transactions based on whether the peer has already seen the transaction, so I think filtering out re-sends on a per-peer basis is okay:

https://github.com/ethereum/devp2p/blob/master/caps/eth.md#transaction-exchange
https://github.com/bitcoin/bitcoin/blob/d1e42659bbdd8da170542d8c638242cd94f71a7d/src/net_processing.cpp#L1898

This change removes the global seenGossipFilter, which fixes the issue where if one peer receives a nonce it has already seen, it won't continue rebroadcasting to its peers. However, we still have to account for the currently-deployed version of Iron Fish. So I believe if we want to do a soft rollout, we wouldn't want to deploy a version that switches over to the new Transaction propagation method until enough people have upgraded to the code in this PR. Does that seem right?

Reply:
Yea you’re right we can’t switch over until enough people have upgraded. I think the main reason for that is that we can’t have that situation where you have to broadcast to a legacy peer and don’t have the original nonce. That would create a bunch of new nonces